presto-main/src/main/java/io/prestosql/operator/TableWriterOperator.java

```diff
@@ -227,7 +236,14 @@ public void addInput(Page page)
 
         Block[] blocks = new Block[columnChannels.size()];
         for (int outputChannel = 0; outputChannel < columnChannels.size(); outputChannel++) {
-            blocks[outputChannel] = page.getBlock(columnChannels.get(outputChannel));
+            Block block = page.getBlock(columnChannels.get(outputChannel));
+            if (block.mayHaveNull()) {
```

Comment:
We should do this inside the check for not-null channels, since this is logically an optimization of checking the block for nulls. For nullable channels, we don't need to look at this.

Reply:
Done in the latest force push.